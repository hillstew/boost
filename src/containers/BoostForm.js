import React, { Component } from 'react';
import { connect } from 'react-redux';
let shortid = require('shortid');
export class BoostForm extends Component {
  constructor() {
    super();
    this.state = {
      senderNum: '',
      recipientNum: '',
      boostMessage: ''
    };
  }

  handleSubmit = e => {
    e.preventDefault();
  };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSave = e => {
    const { dogImgSrc } = this.props;
    const { senderNum, recipientNum, boostMessage } = this.state;
    let boost = {
      id: shortid.generate(),
      senderNum,
      recipientNum,
      boostMessage,
      dogImgSrc
    };
  };

  render() {
    const { dogImgSrc } = this.props;
    const { senderNum, recipientNum, boostMessage } = this.state;
    const url = 'https://random.dog/' + dogImgSrc;
    return (
      <form onSubmit={this.handleSubmit} className="BoostForm">
        <div className="div-flex">
          <div className="input-div">
            <label>
              To:
              <input
                onChange={this.handleChange}
                name="senderNum"
                type="number"
                placeholder="855-546-1234"
                value={senderNum}
              />
            </label>
            <label>
              From:
              <input
                onChange={this.handleChange}
                name="recipientNum"
                type="number"
                placeholder="765-765-2343"
                value={recipientNum}
              />
            </label>
          </div>
          <img src={url} alt="animal" />
        </div>
        <textarea
          onChange={this.handleChange}
          name="boostMessage"
          placeholder="Write your boost here"
          value={boostMessage}
        />
        <button name="save" onClick={this.handleSave}>
          Save
        </button>
        <button name="send" type="submit">
          Send
        </button>
      </form>
    );
  }
}

export const mapStateToProps = state => ({
  dogImgSrc: state.dogImgSrc
});

export default connect(mapStateToProps)(BoostForm);
