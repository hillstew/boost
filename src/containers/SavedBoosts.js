import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removeSaved } from '../actions';

export class SavedBoosts extends Component {
  renderSavedBoosts = () => {
    const { removeSaved, saved } = this.props;
    const url = 'https://random.dog/';
    return saved.map(boost => {
      return (
        <div key={boost.id} className="SavedBoosts-boost">
          <div className="SavedBoosts-flex">
            <div className="div">
              <p>TO: {boost.senderNum}</p>
              <p>FROM: {boost.recipientNum}</p>
            </div>
            <img src={url + boost.dogImgSrc} alt="animal" />
          </div>
          <p className="boost-msg">Your boost messsage: {boost.boostMessage}</p>
          <button name="delete" onClick={() => removeSaved(boost.id)} className="remove-button">
            Delete
          </button>
        </div>
      );
    });
  };

  render() {
    return <div className="SavedBoosts">{this.renderSavedBoosts()}</div>;
  }
}

export const mapStateToProps = state => ({
  saved: state.savedBoosts
});

export const mapDispatchToProps = dispatch => ({
  removeSaved: id => dispatch(removeSaved(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SavedBoosts);
