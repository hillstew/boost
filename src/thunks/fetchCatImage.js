import { setLoading, setError, setCatImage } from '../actions';

export const fetchCatImage = () => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const url = 'https://api.thecatapi.com/v1/images/search';
      const res = await fetch(url);
      if (!res.ok) {
        throw Error(res.statusText);
      }
      dispatch(setLoading(false));
      let catImages = await res.json();
      let image = catImages[0].url;
      image = image.split('/').pop();
      dispatch(setCatImage(image));
    }
    catch (error) {
      dispatch(setError(error.message));
    }
  };
};