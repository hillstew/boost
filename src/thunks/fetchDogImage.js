import { setLoading, setError, setDogImage } from '../actions';
import { randomIndex } from '../utils/helper';

export const fetchDogImage = () => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const url = 'https://random.dog/doggos';
      const res = await fetch(url);
      const dogImages = await res.json();
      const randomImage = dogImages[randomIndex(dogImages.length - 1)];
      dispatch(setDogImage(randomImage));
    } catch (error) {
      console.log(error);
    }
    dispatch(setLoading(false));
  };
};
