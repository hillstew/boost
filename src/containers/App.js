import React, { Component } from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { fetchDogImage } from '../thunks/fetchDogImage';
import { connect } from 'react-redux';
import { Nav } from '../components/Nav';
import { Logo } from '../components/Logo';
import { About } from '../components/About';
import { NotFound } from '../components/NotFound';
import BoostsSection from '../containers/BoostsSection';
import BoostForm from '../containers/BoostForm';
class App extends Component {
  componentDidMount() {
    this.props.fetchDogImage();
  }

  render() {
    const { history } = this.props;
    return (
      <div className="App">
        <div className="App-nav-logo-div">
          <Nav />
          <Logo />
        </div>
        <Switch>
          <Route exact path="/" component={BoostsSection} />
          <Route exact path="/about" component={About} />
          <Route
            path="/send-boost/:imgId"
            render={({ match }) => {
              const { imgId } = match.params;
              return <BoostForm img={imgId} history={history} />;
            }}
          />
          <Route path="*" component={NotFound} />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  dogImgSrc: state.dogImgSrc,
  isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
  fetchDogImage: () => dispatch(fetchDogImage())
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
