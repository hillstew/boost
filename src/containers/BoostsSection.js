import React, { Component } from 'react';
import { BoostCard } from '../components/BoostCard';
import { fetchDogImage } from '../thunks/fetchDogImage';
import { connect } from 'react-redux';

export class BoostsSection extends Component {
  render() {
    const { dogImgSrc, fetchDogImage, isLoading } = this.props;
    return (
      <div>
        {!isLoading && (
          <div className="BoostsSection">
            <h1>Pick your boost image</h1>
            <div className="BoostsSection-cards">
              <BoostCard img={dogImgSrc} fetchDogImage={fetchDogImage} />
            </div>
          </div>
        )}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  dogImgSrc: state.dogImgSrc,
  isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
  fetchDogImage: () => dispatch(fetchDogImage())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BoostsSection);
