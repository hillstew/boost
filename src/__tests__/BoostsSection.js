import React from 'react';
import { shallow } from 'enzyme';
import { BoostsSection } from '../containers/BoostsSection';
import { mapStateToProps } from '../containers/BoostsSection';
import { fetchDogImage } from '../thunks/fetchDogImage';
import { mapDispatchToProps } from '../containers/BoostForm';

jest.mock('../thunks/fetchDogImage');

describe('BoostsSection', () => {
  let wrapper;
  let mockDogImg;
  let mockFetchDog;

  beforeEach(() => {
    mockDogImg = 'rg56-78.jpg';
    mockFetchDog = jest.fn();
    wrapper = shallow(<BoostsSection dogImgSrc={mockDogImg} isLoading={true} fetchDogImage={mockFetchDog} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should reutrn dogImgSrc, isLoading, and hasError', () => {
      const mockState = {
        dogImgSrc: 'fdgsgf',
        isLoading: true,
        extra: 'extra'
      };

      const expected = {
        dogImgSrc: 'fdgsgf',
        isLoading: true
      };

      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when fetchDogImage is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchDogImage();
      const props = mapDispatchToProps(mockDispatch);
      props.fetchDogImage();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
