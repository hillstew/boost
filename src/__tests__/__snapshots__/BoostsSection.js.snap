// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoostsSection should match the snapshot 1`] = `<div />`;
